base/strings/util.jl

```diff
@@ -77,8 +77,25 @@ julia> a = "March"
 
 julia> chop(a)
 "Marc"
+
+julia> chop(a, 1, 2)
+"ar"
+
+julia> chop(a, 5, 5)
+""
 ```
 """
+function chop(s::AbstractString, head::Integer, tail::Integer)
+    # negative values of head/tail will throw error in nextind/prevind
+    if head == 0
+        tail == 0 && return SubString(s)
+        return SubString(s, 1, prevind(s, endof(s), tail))
```

Comment:
AFAIK we use `start(s)` instead of `1` in the existing code, though I suspect we should move to `1` everywhere.

Reply:
changed to `start`. There is a mixed style in the code base (I have seen both during the last month :)), but I agree that it is better to use `start`. I fix some more situations like this in `strings/util.jl`.